syntax = "proto3";

package pagamento;

// Mensagens para Pagamento
message ProcessarPagamentoRequest {
    int32 leilao_id = 1;
    string cliente_id = 2;
    double valor = 3;
    string moeda = 4;
}

message ProcessarPagamentoResponse {
    bool success = 1;
    string message = 2;
    string id_transacao = 3;
    string link_pagamento = 4;
}

message NotificacaoPagamento {
    string tipo = 1; // "link_pagamento", "status_pagamento"
    int32 leilao_id = 2;
    string cliente_id = 3;
    string id_transacao = 4;
    string link_pagamento = 5;
    string status = 6;
    double valor = 7;
}

// Servi√ßo de Pagamento
service PagamentoService {
    rpc ProcessarPagamento(ProcessarPagamentoRequest) returns (ProcessarPagamentoResponse);
}
